### PR TITLE
Update typescript guide

### PR DESCRIPTION
**Pull request by mgrybyk**
## Proposed changes

* recommend using typed configuration as it's more safe and more easier to setup
* remove `tsconfig-paths` to reduce complexity
* use `transpileOnly` option in `ts-node` to avoid issues with custom types, see #5766 
* add `tsconfig.json` example

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

The PR is draft due to the discussion in https://github.com/webdriverio/webdriverio/issues/5766

### CODE REVIEW

File: docs/TypeScript.md [hunk 1]
```diff
@@ -222,3 +200,38 @@ declare module WebdriverIO {
...
 ```
 <!--END_DOCUSAURUS_CODE_TABS-->
+
+## Tips and Hints
+
+### tsconfig.json example
+It's still good to have an example
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--Mocha + Sync Mode-->
```
Comment (christian-bromann) [inline docs/TypeScript.md:210]: No need for code tabs if there is only one example

```suggestion
```

File: docs/TypeScript.md [hunk 2]
```diff
@@ -222,3 +200,38 @@ declare module WebdriverIO {
...
+      "@wdio/mocha-framework",
+      "@wdio/sync"
+    ],
+  },
+  "include": [
+    "./test/**/*.ts"
+  ]
+}
+```
+<!--END_DOCUSAURUS_CODE_TABS-->
```
Comment (christian-bromann) [inline docs/TypeScript.md:233]:
```suggestion
```

**Review by christian-bromann — APPROVED** [review]
Overall looks good to me, one suggestion.

File: docs/TypeScript.md (hunk 2 above)
Comment (mgrybyk) [reply docs/TypeScript.md]: done

File: docs/TypeScript.md [hunk 3]
```diff
@@ -222,3 +200,35 @@ declare module WebdriverIO {
 }
 ```
 <!--END_DOCUSAURUS_CODE_TABS-->
+
+## Tips and Hints
+
+### tsconfig.json example
+It's still good to have an example
```
Comment (christian-bromann) [inline docs/TypeScript.md:207]: Is this sentence intentional?

File: docs/TypeScript.md [hunk 4]
```diff
@@ -144,27 +130,17 @@ For instance, if you decide to use the Mocha framework, you need to install `@ty
...
-        },
         "types": ["node", "webdriverio", "@wdio/jasmine-framework"]
     },
     "include": [
         "./test/**/*.ts"
     ]
 }
 ```
-<!--Cucumber-->
+<!--Cucumber -->
```
Comment (christian-bromann) [inline docs/TypeScript.md:140]: this change seems unnecessary
```suggestion
<!--Cucumber-->
```

File: docs/TypeScript.md (hunk 3 above)
Comment (mgrybyk) [reply docs/TypeScript.md:207]: Haha, yes. However I agree that it makes no sense, removing.

**Review by christian-bromann — APPROVED** [review]
👍

**Pull request merged**